### PR TITLE
Refactor e2e tests

### PR DESCRIPTION
**Pull request by joaquim-verges**
## Changes

- added tests for contract write
- using prool to start/stop anvil instance
- fixes to ensure custom chains are respected all the way through
- fix for userop type error
- sdk upgrade

## How this PR will be tested

`bun test test_name`

---

## PR-Codex overview
This PR focuses on updating dependencies, enhancing contract interaction methods, and improving error handling in the codebase. It also includes the addition of new functionality related to contract operations and adjustments to the testing setup.

### Detailed summary
- Updated `thirdweb` dependency from `^5.56.0` to `^5.58.4`.
- Removed `viem.ts` and `chain.ts` files.
- Changed `getContractV5` to be an `async` function in multiple files.
- Added `outputs` to `abiSchema` in `src/server/schemas/contract/index.ts`.
- Introduced `internalType` in the `abiSchema`.
- Enhanced error handling in `writeToContract` function.
- Updated `setup` function in tests to use `startAnvil` and `stopAnvil`.
- Added new private keys and client setup in `test/e2e/utils/wallets.ts`.
- Improved the structure of the `CONFIG` object in `test/e2e/config.ts`.
- Added multiple tests for contract interactions in `test/e2e/tests/write.test.ts`.

> The following files were skipped due to too many changes: `test/e2e/yarn.lock`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: test/e2e/package.json [hunk 1]
```diff
@@ -10,9 +10,7 @@
   },
   "type": "module",
   "dependencies": {
-    "@thirdweb-dev/sdk": "^4.0.99",
-    "ethers": "5",
-    "thirdweb": "^5.43.2",
-    "viem": "^2.18.1"
```
Comment (d4mr) [inline test/e2e/package.json:16]: We want to have tests that turn off interval mining on the chain to simulate features like the chain stopping producing blocks etc, and [viem test client](https://viem.sh/docs/actions/test/mine.html) is helpful there.

File: test/e2e/utils/engine.ts [hunk 2]
```diff
@@ -59,15 +60,3 @@ export const getEngineBackendWallet = async (engine: Engine) => {
   console.log("Using existing wallet", res.result[0].address);
   return getAddress(res.result[0].address);
 };
-
-export const setWalletBalance = async (
```
Comment (d4mr) [inline test/e2e/utils/engine.ts:63]: thoughts on being able to explicitly set the balance on an account vs depending on the default anvil accounts to have balance?

File: test/e2e/tests/setup.ts [hunk 3]
```diff
@@ -26,25 +22,26 @@ export const setup = async (): Promise<SetupResult> => {
...
 
-  const publicClient = setupPublicClient();
   const engine = setupEngine();
   const backendWallet = await getEngineBackendWallet(engine);
-  let testClient;
+  await engine.backendWallet.resetNonces();
 
   if (CONFIG.USE_LOCAL_CHAIN) {
-    setupAnvil();
+    startAnvil();
```
Comment (d4mr) [inline test/e2e/tests/setup.ts:30]: if we `await startAnvil()` we can probably get rid of the `sleep(1000)`, I assume prool helps us know when anvil has started producing blocks?

**Review by d4mr — APPROVED** [review]

File: test/e2e/package.json (hunk 1 above)
Comment (joaquim-verges) [reply test/e2e/package.json:16]: I think you can configure that with prool, but can always add back!

File: test/e2e/utils/engine.ts (hunk 2 above)
Comment (joaquim-verges) [reply test/e2e/utils/engine.ts:63]: Again didn't need it so rather keep things minimal. Can always add more on top

File: test/e2e/tests/setup.ts (hunk 3 above)
Comment (joaquim-verges) [reply test/e2e/tests/setup.ts:30]: Yep good point, will add as a follow up

**Pull request merged**